Google.PowerShell/BigQuery/BqJob.cs

```diff
@@ -178,5 +178,257 @@ public void DoGetRequest()
+        /// invalid, an error returns. //This feature is not currently available.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.DoQuery)]
+        public SwitchParameter DryRun { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Specifies BigQuery's legacy SQL dialect for this query.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.DoQuery)]
+        public SwitchParameter UseLegacySql { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// A dataset to use for any unqualified table names in the query.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.DoQuery)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Dataset),
+            Property = nameof(Dataset.DatasetReference))]
+        [Alias("Dataset")]
+        public DatasetReference DefaultDataset { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The destination table to write the results into. If this is not specified, the 
+        /// results will be stored in a temporary table.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.DoQuery)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Table),
+            Property = nameof(Table.TableReference))]
+        [Alias("Table", "Dest")]
+        public TableReference DestinationTable { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Priority of the query.  Can be 'Batch' or 'Interactive'.
+        /// </para>
+        /// </summary>
+        public QueryPriority Priority { get; set; }
+
+        // Copy Parameters.
+
+        /// <summary>
+        /// <para type="description">
+        /// Selects job type Copy.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.DoCopy)]
+        public SwitchParameter Copy { get; set; }
+
+        // Load Parameters.
+
+        /// <summary>
+        /// <para type="description">
+        /// Selects job type Load.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.DoLoad)]
+        public SwitchParameter Load { get; set; }
+
+        // Extract Parameters.
+
+        /// <summary>
+        /// <para type="description">
+        /// Selects job type Extract.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.DoExtract)]
+        public SwitchParameter Extract { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            var Client = BigQueryClient.Create(Project);
```

Comment:
In our Apiary based cmdlets put the Client in an Inherited property. I suggest you do the same here (possibly making a separate abstract subclass).